hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -78,14 +81,29 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
       // setup configuration
       long ckpTimeout = dataStream.getExecutionEnvironment()
           .getCheckpointConfig().getCheckpointTimeout();
-      conf.setLong(FlinkOptions.WRITE_COMMIT_ACK_TIMEOUT, ckpTimeout);
+      conf.set(FlinkOptions.WRITE_COMMIT_ACK_TIMEOUT, ckpTimeout);
       // set up default parallelism
       OptionsInference.setupSinkTasks(conf, dataStream.getExecutionConfig().getParallelism());
       // set up client id
       OptionsInference.setupClientId(conf);
       // set up index related configs
       OptionsInference.setupIndexConfigs(conf);
 
+      // Since Flink 2.0, the adaptive execution for batch job will generate job graph incrementally
+      // for multiple stages (FLIP-469). And the write coordinator is initialized along with write
+      // operator in the final stage, so hudi table should be initialized if necessary during the plan
+      // compilation phase when adaptive execution is enabled.
```

Comment:
don't quite understand this, what happens in batch execution if it is not flink2?